src/main/java/org/testng/SuiteRunner.java

```diff
@@ -220,9 +247,17 @@ public int compare(XmlTest arg0, XmlTest arg1) {
       }
     });
 
+    boolean isNewVariant = (iTestRunnerFactory instanceof ITestRunnerFactory2);
```

Comment:
Fixed the usage of the boolean. But I didn't understand the proxy part that you are talking about and why we don't need the extension. Can you please help me understand that ?